### ISSUE TITLE
RFE: generate a sourcemap of what code affects which tests

### ISSUE THREAD

**Issue by mathstuf** [sourcefrog/cargo-mutants#427]
It would be nice to be able to generate some kind of database that records what tests cover which source lines. This could then be consumed by something like `nextest` in conjunction with `git diff` to only run tests that are affected by the changes in a patch to avoid running the entire test suite all the time.

**Comment by mathstuf** [comment]
As a bonus, CI could also mutate just the diff and run tests in that direction too.

**Comment by sourcefrog** [comment]
This is similar to #24.

But, maybe here you're saying we could remember which specific tests failed due to each specific mutation?

**Comment by mathstuf** [comment]
> But, maybe here you're saying we could remember which specific tests failed due to each specific mutation?

Yes. When discussing with colleages, they suggested doing it by coverage, but I'm not aware of any coverage tools that save such information in a per-test bucket (especially in-process testing like Rust's `#[test]` or `pytest`).

**Comment by sourcefrog** [comment]
Yes, I think it's an interesting idea. I think it's more closely connected to a coverage tool than to cargo-mutants, and it would probably require running each test separately like Nextest does.
